データベース設計

AG0811 opened the pull request:
＃What
「.dio」にてER図を作成
READMEにてデータベース設計図を作成
＃Why
データベース作成のため、関連付けやイメージしやすいように可視化ため

◆ER図(Gyazo)：
https://gyazo.com/1d5cd6c250c24df8a46618b08242b35d
※補足「has_many  | ← |  belongs_to」「has_one  |- |  」

AG0811 committed:
Rubocopを実行

AG0811 committed:
データベース設計

＃What
[.dio]にてER図作成
READMEにてDB設計を記述
＃Why
データベース作成のためのデータベース設計

TakashiKozuma commented on README.md:
```diff
@@ -1,24 +1,69 @@
-* Ruby version
+has_many : items
+has_many : transactions
+has_many :addresses
```
```Rb
has_many :addresses
```
住所情報とのアソシエーションは削除しましょう。
必須機能の実装では住所情報は購入のたびに入力する仕様となっています。
そのため、住所はユーザーと結びつけるのではなく、購入履歴とのみ結びつけるようにする必要があります。

TakashiKozuma commented on README.md:
```diff
@@ -1,24 +1,69 @@
 
-* Database creation
+belongs_to : user
+belongs_to : transaction
```
```Rb
belongs_to : transaction
```
購入機能の方が後から生成されるので、belongs_toではなくhas_oneに修正しましょう。

TakashiKozuma commented on README.md:
```diff
@@ -1,24 +1,69 @@
-* ...
+| Column         | Type       | Options                        |
+| -------------- | ---------- | ------------------------------ |
+| user           | references | null: false, foreign_key: true |
```
```Rb
| user | references | null: false, foreign_key: true |
```
今回の実装ではユーザーの外部キーは削除しましょう。

TakashiKozuma commented on README.md:
```diff
@@ -1,24 +1,69 @@
+| building_name  | string     |                                |
+| phone_number   | string     | null: false                    |
+
+belongs_to :user
```
```Rb
belongs_to :user
```
ユーザーとのアソシエーションのみ削除しましょう。

AG0811 replied:
レビューありがとうございます。
上記の旨かしこまりました。
修正後再度レビュー依頼をさせていただきます。

TakashiKozuma reviewed (commented):
レビュー依頼ありがとうございます！
確認させていただきました。
コメントを致しましたので、ご確認お願いします。
また、次回レビュー依頼時に最新状態のER図の添付もお願いいたします。

修正後はコミット・プッシュのし忘れがないかご確認の上、再度フォームからレビュー依頼をお願い致します。

AG0811 committed:
ER図修正

usersテーブル：has_many :addresses消去
itemsテーブル：has_one：transactionに修正
addressesテーブル：user消去、belongs_to :user消去

AG0811 committed:
ER図修正

ER図修正

AG0811 commented:
＃What【修正】
usersテーブル：has_many :addresses消去（購入履歴とのみ結び付きのため）
itemsテーブル：has_one：transactionに修正（購入機能と1対1のため）
addressesテーブル：user消去、belongs_to :user消去（購入機能からの遷移のため）

＃Why【修正】
データベース設計の修正のため

◆【修正】ER図
https://gyazo.com/a6394a41c4f8a3aa2549233e623aa20d

TakashiKozuma commented:
追加のレビュー依頼ありがとうございます！
修正点も含め、確認が完了しました。
LGTMです！！

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に次の実装のためのブランチを切ってから次にお進み下さい。
